### ISSUE TITLE
Mechanism for identifying active members

### ISSUE THREAD

**Issue by fowles** [nysenate/OpenLegislation#90]
Consider. https://legislation.nysenate.gov/api/3/members/2022/senate?limit=1000&key=REDACTED

this responds with a large list that contains

```
{
      "sessionMemberId" : 1864,
      "shortName" : "CLEARE",
      "sessionYear" : 2021,
      "districtCode" : 30,
      "alternate" : false,
      "memberId" : 1504
},
```
and
```
{
      "sessionMemberId" : 1798,
      "shortName" : "BENJAMIN",
      "sessionYear" : 2021,
      "districtCode" : 30,
      "alternate" : false,
      "memberId" : 1136
}, 
```

CLEARE is the active member having replaced BENJAMIN in a special election, but the API does not provide a way of determining that.

Observationally, the `memberId` and `sessionMemberId` are larger, which is likely to do with them being handed out in order, but this is not documented anywhere.  Ideally, there would be a documented way to determine the active member in cases like this.  I propose adding a boolean `"active" : true` which will be true for the member that is currently serving and `false` for a member that has been replaced.

**Comment by KevinCaseiras** [comment]
Hey @fowles,

There are a few options available to determine who is active but it looks like they are missing from our docs.

One option is to add the `full=true` search param. i.e. `/api/3/members/2022/senate?limit=1000&full=true` This will give you a response with an `incumbent` field, which is true for currently serving members.
```
{
  "memberId": 1504,
  "chamber": "SENATE",
  "incumbent": true,
  "fullName": "Cordell Cleare",
  "shortName": "CLEARE",
  "imgName": "Cordell Cleare.jpg",
  "sessionShortNameMap": {
    "2021": [
      {
        "sessionMemberId": 1864,
        "shortName": "CLEARE",
        "sessionYear": 2021,
        "districtCode": 30,
        "alternate": false,
        "memberId": 1504
      }
    ]
  },
  "person": {
    "personId": 1321,
    "fullName": "Cordell Cleare",
    "firstName": "Cordell",
    "middleName": "",
    "lastName": "Cleare",
    "email": "",
    "prefix": "Senator",
    "suffix": "",
    "verified": true,
    "imgName": "Cordell Cleare.jpg"
  },
  "alternate": false,
  "sessionMemberId": 1864,
  "sessionYear": 2021,
  "districtCode": 30
}
```

Another option is to use our search api. A call to 
`/api/3/members/search?term=_exists_:sessionShortNameMap.2021%20AND%20incumbent:true` will give you a summary of all currently active members.

Our member data is updated manually, so at times it may be missing data or slightly out of date. If your looking for the most accurate and up to date info on current senators, https://www.nysenate.gov/senators-json may be a better source of data for you (unfortunately I'm not aware of a similar api for assembly members).

I'm going to close this issue, but if you have any other issues or questions feel free to re open this or create a new issue.

**Comment by fowles** [comment]
When I try this with full=true I see

```
      "memberId" : 384,
      "chamber" : "SENATE",
      "incumbent" : false,
      "fullName" : "Joseph P. Addabbo Jr.",
      "shortName" : "ADDABBO",
      "imgName" : "384_joseph_p._addabbo_jr..jpg",
```

but he is still an active member of the senate (I am pretty sure).

Is there a data error?

**Comment by fowles** [comment]
I don't appear able to reopen this issue

**Comment by KevinCaseiras** [comment]
Yes that does appear to be a data error, thanks for pointing it out.
